lib/rosette/core/branch_utils.rb

```diff
@@ -0,0 +1,131 @@
+# encoding: UTF-8
+
+require 'digest/sha1'
+
+module Rosette
+  module Core
+
+    # Utility methods that work at the branch level.
+    class BranchUtils
+      class << self
+
+        # Computes the status for an array of commit logs. Determining the
+        # aggregate status of more than one commit log means choosing the least
+        # far-along status from the list. In other words, if the list contains
+        # two commit logs with UNTRANSLATED and TRANSLATED statuses
+        # respectively, this method will return UNTRANSLATED because that's the
+        # least advanced status in the list. A list of commit logs is only as
+        # advanced as its 'weakest', least advanced member.
+        #
+        # @param [Array<CommitLog>] commit_logs The list of commit logs to
+        #   derive the status from.
+        # @return [PhraseStatus]
+        def derive_status_from(commit_logs)
+          ps = Rosette::DataStores::PhraseStatus
+          entry = commit_logs.min_by { |entry| ps.index(entry.status) }
+
+          if entry
+            entry.status
+          else
+            # since TRANSLATED commit_logs aren't considered, the absence of
+            # any commit logs (i.e. a nil entry) indicates the branch is
+            # fully translated
+            Rosette::DataStores::PhraseStatus::TRANSLATED
+          end
+        end
+
+        # Computes the phrase count for the given list of commit logs. This is
+        # simply the sum of all the commit log's phrase counts.
+        #
+        # @param [Array<CommitLog>] commit_logs The list of commit logs to
+        #   compute the phrase count from.
+        # @return [Fixnum]
+        def derive_phrase_count_from(commit_logs)
+          commit_logs.inject(0) { |sum, entry| sum + entry.phrase_count }
```

Comment:
not super important, but `inject` defaults to using the first element of the array if you don't supply an initial value